### PR TITLE
Add oauth2 owner flow support

### PR DESCRIPTION
**Pull request by mhum**
This is work for #19.

@evert,
I've added a new auth type: `oauth2` and support for the owner flow. What I've done so far is to just request the authorization token when it is missing and set it in the ketting object. You can see in the new tests how configuring this works. This makes use of https://github.com/mulesoft/js-client-oauth2 for managing the oauth client.

I still need to add support for refreshing the token and updating the docs, but I'd like to get your initial thoughts.

### CODE REVIEW

File: lib/utils/oauth.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+   * Fetches an oauth2 token and sets it on the given ketting class
+   *
+   * @async
+   * @param {object} ketting - Ketting object.
+   * @return {object}
+   */
+  fetchToken: function(ketting) {
+    if (ketting.auth.owner) {
+      return this.ownerFlow(ketting);
+    }
```
Comment (evert) [inline lib/utils/oauth.js]: An exceptioin for the else clause would be nice

File: lib/utils/oauth.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   * @param {object} ketting - Ketting object.
+   * @return {object}
+   */
+  ownerFlow: function(ketting) {
+    return ketting.auth.oauthClient.owner.getToken(
+      ketting.auth.owner.userName,
+      ketting.auth.owner.password
+    )
+      .then(function (client) {
+        ketting.auth['accessToken'] = client.accessToken;
```
Comment (evert) [inline lib/utils/oauth.js]: This could be `ketting.auth.accessToken`

**Comment by mhum** [comment]
Just need to update the docs before this is completely ready to go.

**Comment by evert** [comment]
K ! feel free to merge if you feel good about it, and let me know if you need me to publish a new release on npm.

**Comment by mhum** [comment]
Can you have a look at the updated REAME to make sure it makes sense and fits in with the overall flow and style?

**Comment by evert** [comment]
Still looks great. Thanks for correcting my typos

**Pull request merged**